### PR TITLE
Adds an alternative method to creating black carpet

### PR DESCRIPTION
**Pull request by Goodstuff123456**
:cl: Goodstuff
Add: Black carpet is can now be made by hitting a crayon against regular carpet
tweak: The black carpet recipe in the crafting menu gives less black carpet
/:cl:

Now you can make a few extra tiles of black carpet without needing to go and steal the library floor for just 3 extra black carpet tiles.

### CODE REVIEW

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 1]
```diff
@@ -112,8 +112,15 @@
 	icon_state = "tile-carpet-black"
 	turf_type = /turf/open/floor/carpet/black
 
-/obj/item/stack/tile/carpet/black/fifty
-	amount = 50
+/obj/item/stack/tile/carpet/attackby(obj/item/O, mob/user, params)
+	if (istype(O, /obj/item/toy/crayon/black))
+		amount -= 1
```
Comment (Expletive) [inline code/game/objects/items/stacks/tiles/tile_types.dm]: `if (istype(O, /obj/item/toy/crayon/black))
		use(1)
		new/obj/item/stack/tile/carpet/black(user.loc)
 ..()`

**Comment by Expletive** [comment]
Also, this is great. Crafting datum are for lazy people like me.

File: code/game/objects/items/stacks/tiles/tile_types.dm (hunk 1 above)
Comment (Goodstuff123456) [inline code/game/objects/items/stacks/tiles/tile_types.dm]: wait what thats a thing

Comment (Goodstuff123456) [inline code/game/objects/items/stacks/tiles/tile_types.dm]: oh hey that does work

**Comment by Cheridan** [comment]
so making people click 50 times is an improvement??

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 2]
```diff
@@ -114,13 +114,9 @@
...
 	if (istype(O, /obj/item/toy/crayon/black))
-		amount -= 1
+		use(1)
 		new/obj/item/stack/tile/carpet/black(user.loc)
-		if(amount == 0)
-			del src
-		else
-	else
-		return ..()
+		..()
```
Comment (Expletive) [inline code/game/objects/items/stacks/tiles/tile_types.dm]: The ..() is inside the if statement. It shouldn't be.

**Comment by Goodstuff123456** [comment]
Can you not spend 7 seconds clicking on a stack with a crayon? Also with the old way you had to spend time making 50 extra carpet tiles if you only needed 3 extra tiles. How is this not an improvement?

**Comment by Expletive** [comment]
Cheridan is also probably right. Maybe leave in the crafting menu option.  Also, possibly  subtract one from the crayon's charges var after you create a black carpet.

**Comment by Goodstuff123456** [comment]
Not a fan of the crafting menu but crayon charges sounds like a good idea.

**Comment by Expletive** [comment]
Closing this? Disappointing.

**Comment by Goodstuff123456** [comment]
reopening it later

**Comment by FantasticFwoosh** [comment]
You forgot about something called spraypaint @Cheridan , might as well let it change the carpet colour rather than a ugly colour overlay ontop of the tile if drawing on it is going to have a effect.

**Pull request closed without merging**